grails-app/services/com/unifina/service/EthereumIntegrationKeyService.groovy

```diff
@@ -81,24 +83,28 @@ class EthereumIntegrationKeyService {
 		if (invalidChallenge) {
 			throw new ApiException(400, "INVALID_CHALLENGE", "challenge validation failed")
 		}
+
 		def message = challenge
-		byte[] address;
+		String address
 		try {
 			byte[] messageHash = ECRecover.calculateMessageHash(message)
 			address = ECRecover.recoverAddress(messageHash, signature)
 		} catch (SignatureException | DecoderException e) {
 			throw new ApiException(400, "ADDRESS_RECOVERY_ERROR", e.message)
 		}
-		IntegrationKey key = new IntegrationKey(
+
+		if (IntegrationKey.findByServiceAndIdInService(IntegrationKey.Service.ETHEREUM_ID, address) != null) {
+			throw new DuplicateNotAllowedException("Ethereum ID already taken.")
```

Comment:
How about: "This Ethereum address is already associated with another Streamr user".